src/main/java/me/juancarloscp52/bedrockify/client/features/hudOpacity/HudOpacity.java

```diff
@@ -2,32 +2,20 @@
 
 import me.juancarloscp52.bedrockify.client.BedrockifyClient;
 import net.minecraft.client.MinecraftClient;
+import net.minecraft.util.math.MathHelper;
 
 public class HudOpacity {
 
-    private final int maxInactiveTicks = 120; // 6 seconds.
+    private static final int maxInactiveTicks = 120; // 6 seconds.
+    private static final int FADE_OUT_START_TICK = 100;
+    private static final int FADE_IN_DURATION_TICK = 5;
     int inactiveTicks = 0;
     int previousSelectedSlot = -1;
-    boolean hasReachedMax = false;
+    float currentOpacity = 1f;
 
     public float getHudOpacity(boolean isHotBarBackground){
-        float opacity = BedrockifyClient.getInstance().settings.hudOpacity/100f;
         float max = BedrockifyClient.getInstance().settings.isTransparentHotBarEnabled() && isHotBarBackground? 0.6F:1.0F;
-
-        if(opacity>=max-0.05)
-            return max;
-
-        float delta = max - opacity;
-        if(inactiveTicks > 100){
-            hasReachedMax=true;
-            return max - (delta * ((inactiveTicks-100f)/20f));
-        }else{
-            if(inactiveTicks<=5 && hasReachedMax){
-                return opacity + (delta * ((inactiveTicks)/5f));
-            } else
-                hasReachedMax=false;
-            return max;
-        }
+        return Math.min(currentOpacity, max);
```

Comment:
From Curseforge comment:

> hotbar transparency to disable it does not work well in 1.19.4

Is it better to prioritize the Transparent HotBar setting? Such as turning it OFF would ignore the Hud Opacity functionality.

Reply:
Hello! I'm not able to replicate the problem from the comment of not being able to disable hot bar transparency with the latest version of bedrockIfy (and also cannot reproduce the error in this PR). Seems to be working as expected, being able to disable hot bar opacity, but hud opacity continues working, and vice versa.  Maybe I am missing something? If the error does not longer exists, we don't need to worry about spliting HUD and hotbar opacity. BUT, if we are able to reproduce it and is unintended, your solution in https://github.com/lonefelidae16/BedrockIfy/commit/e018f065214f8ea91db474e21f42a220ef67a691 is great.

The PR is awesome, thank you for solving the block item opacity :)

Reply:
I'm going to merge this PR and if we end up being able to reproduce the issue, we can merge that fix after wards ;)